library/src/main/java/com/getstream/sdk/chat/view/ReadStateView.java

```diff
@@ -82,7 +81,7 @@ private void init() {
         }
         // Count Text
         TextView textView = new TextView(getContext());
-        textView.setText(String.valueOf(reads.size()));
+        textView.setText(String.valueOf(reads.size() - 1));
```

Comment:
- does `read` always contain current user? if not then this count will be wrong?
- what about empty read state? this will set the text to -1

Reply:
> does read always contain current user? if not then this count will be wrong?
`read` doesn't contain always current user.
Please see this line
https://github.com/GetStream/stream-chat-android/blob/master/library/src/main/java/com/getstream/sdk/chat/utils/MessageListItemLiveData.java#L89

```java
        ...
        // this wil become slow with many users and many messages
        for (Map.Entry<String, ChannelUserRead> entry : readsByUser.entrySet()) {
            // we don't show read state for the current user
            if (entry.getValue().getUser().getId().equals(currentUser.getId())) {
                continue;
            }
        ...
```

> what about empty read state? this will set the text to -1
There won't be this case. ;)
Please see this  line
https://github.com/GetStream/stream-chat-android/blob/master/library/src/main/java/com/getstream/sdk/chat/view/ReadStateView.java#L49

```java
if (!style.isShowReadState()
    || reads == null
    || reads.isEmpty()) return;
```